parser/prism/Translator.cc

```diff
@@ -1337,6 +1351,53 @@ unique_ptr<parser::Node> Translator::translateCallWithBlock(pm_block_node *prism
     return make_unique<parser::Block>(sendNode->loc, move(sendNode), move(blockParametersNode), move(body));
 }
 
+// Prism represents a `begin ... rescue ... end` construct using a `pm_begin_node` that may contain:
+//   - `statements`: the code before the `rescue` clauses (the main body).
+//   - `rescue_clause`: a `pm_rescue_node` representing the first `rescue` clause.
+//   - `else_clause`: an optional `pm_else_node` representing the `else` clause.
+//
+// Each `pm_rescue_node` represents a single `rescue` clause and is linked to subsequent `rescue` clauses via its
+// `consequent` pointer. Each `pm_rescue_node` contains:
+//   - `exceptions`: the exceptions to rescue (e.g., `RuntimeError`).
+//   - `reference`: the exception variable (e.g., `=> e`).
+//   - `statements`: the body of the rescue clause.
+//
+// In contrast, Sorbet's legacy parser represents the same construct using a `Rescue` node that contains:
+//   - `body`: the code before the `rescue` clauses (the main body).
+//   - `rescue`: a list of `Resbody` nodes, each representing a `rescue` clause.
+//   - `else_`: an optional node representing the `else` clause.
+//
+// This function and the PM_BEGIN_NODE case translate between the two representations by processing the `pm_rescue_node`
+// (and its linked `subsequent` nodes) and assembling the corresponding `Rescue` and `Resbody` nodes in Sorbet's AST.
+unique_ptr<parser::Node> Translator::translateRescue(pm_rescue_node *prismRescueNode, unique_ptr<parser::Node> bodyNode,
+                                                     unique_ptr<parser::Node> elseNode) {
+    NodeVec rescueBodies;
+
+    // Loop through all rescue clauses, starting from the initial one and going through each subsequent one.
```

Comment:
I'd remove this comment, as it's pretty much verbatim what the `for` loop is already saying